binderhub/repoproviders.py

```diff
@@ -83,6 +106,135 @@ def get_build_slug(self):
         return '{user}-{repo}'.format(user='Rick', repo='Morty')
 
 
+class BasicGitRepoProvider(RepoProvider):
+    """Bare bones git repo provider.
+
+    Users must provide a spec and has to specify a resolved_ref as a query parameter.
+
+    eg:
+    /build/git/https%3A//git.example.com/ns/repo.git?resolved_ref=a01ce557f868d7060a6099bd3c2d739734dcaf15
```

Comment:
Can we instead just use:

```
/build/git/<url-encoded-url>/<full-resolved-sha>
```

to keep consistent with the github provider? While not codified anywhere, I'd like us to reserve query parameters only for runtime parameters (that can change after build) rather than build parameters.